detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -303,6 +337,43 @@ class AndroidDriver extends DeviceDriverBase {
   _getAdbName(deviceId) {
     return _.isObjectLike(deviceId) ? deviceId.adbName : deviceId;
   }
+
+  async _waitForAppLaunch(adbName, bundleId, launchArgs) {
+    const instrumentationClass = await this.adb.getInstrumentationRunner(adbName, bundleId);
+    this._printInstrumentationHint({ instrumentationClass, launchArgs });
+    await pressAnyKey();
+    await this._reverseServerPort(adbName);
+
+    // TODO: think about a more elegant way to bypass instrumentation
+    this.instrumentation.waitForCrash = async () => {};
```

Comment:
@d4vidi, any idea how to improve here in a way true to your instrumentation helper object design?

Reply:
@noomorph if `instrumentation` is not at play here (test is instrumented manually), can the call to do [this function](https://github.com/wix/Detox/pull/2540/files#diff-1f95bce1adbbe6aae44166a89a62433fba2b9bb8d98ab439342abd509f8ff937R278) be omitted altogher, instead?

Reply:
@d4vidi , I am not sure if I am calling that function at all, in the manual flow.

I'd say that the problem stems rather from the fact that no one calls instrumentation init, while later we call this place:
```
        await Promise.race([super.waitUntilReady(), this.instrumentation.waitForCrash()]);
```

and non-initialized instrumentation throws an error when waitingForCrash - because it cannot find the process id of the instrumentation running, which is obvious because we never wanted to launch it in the first place.

Reply:
Well off the top of my head I can say that waiting for crash isn't all that relevant here either. Since detox doesn't launch instrumentation, it should not account for its crashing. I think it's safe to say that in this manual mode, the user is hands on with these things.

Reply:
Okay, I see what you mean. Correct!
I think I know what to do. Thanks.